docs/Performance/promise.md

```diff
@@ -0,0 +1,195 @@
+---
+layout: default
+title: Promise
+name: Overview
+parent: Performance
+nav_order: 1
+---
+
+## Promise
+
+If you are new to Promises, I would suggest to make yourself familiar before continuing:
+
+<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise>
+
+### Basic OData call
+
+This is a very basic OData call where the key and the callbacks are hardcoded.
+
+```js
+var model = that.getView().getModel();
+model.read("/MyEntitySet('12345')", {
+  success: function (data, response) {
+    // success handler
+  },
+  error: function (error) {
+    // error handler
+  }
+});
+```
+
+### Promisified OData call
+
+A wrapped version of the basic OData call.
+
+The key is given as parameter and the wrapping method returns a promise. The resolve / reject methods are used in the success / error callbacks to resolve or reject the Promise.
+
+As the resolve method takes only one parameter, data and response are passed as object.
+
+```js
+function readMyEntity(id) {
+  return new Promise(function (resolve, reject) {
+    var model = that.getView().getModel();
+    model.read("/MyEntitySet('" + id + "')", {
+      success: function (data, response) {
+        resolve({ data: data, response: response });
+      },
+      error: function (error) {
+        reject(error);
+      }
+    });
+  });
+}
+```
+
+The success and error handling can now be done after invoking the method.
+
+```js
+readMyEntity(12345)
+  .then(function (result) {
+    // success handler
+    console.log(result.data, result.response);
+  })
+  .catch(function (error) {
+    // error handler
+    console.error(error);
+  });
+```
+
+### Refactor the Promisified OData call
+
+Now, let's make the OData call more robust for everyday usage:
+
+* hand over the whole entity as object and construct the key from its data
+* provide an option to add additional data like filters to the call (mParameters as described in the [documentation](https://ui5.sap.com/#/api/sap.ui.model.odata.v2.ODataModel%23methods/read))
+* make sure its not possible to provide success / error callbacks with the parameters object as this would break the promise functionality
+
+The UI5 framework detects when the same OData call is done multiple times. It does only one request and returns the same result object for every call. If this result object is being changed, it changes for all calls. To prevent this from happening, the result is being copied with Object.assign()
```

Comment:
I'm pretty sure you might have gained that knowledge by painful (debugging) experiences :) ...
but are you 100% sure about this? That it's UI5 "caching" OData reads and not the browser itself? Or even a `http 304` on the backend/server-side causing this?

Reply:
Yes, I just verified it. 
If you want, you can verify it on your own.
Place the demo methods in a controller and update the entity name.
I results in just one backend call, it doesn't mater if batch is set to true or not.
The 2nd call gets the changed result value.

However, if you use the commented out callback call in the readEntity method, it gets the value as expected.

```js
  readEntity: function (callback) {
    this.getView()
      .getModel()
      .read('/YourEntity', {
        success(result) {
          // callback(Object.assign({}, result))
          callback(result)
        }
      })
  },
  onInit: function () {
    this.readEntity(result => {
      console.log('demo1', result)
      result.results = 'demo1'
      console.log('demo1', result)
    })
    this.readEntity(result => {
      console.log('demo2', result)
    })
```